### ISSUE TITLE
Any way of modifying markers or adding data to ClusterMarker ?

### ISSUE THREAD

**Issue by Twansklaf** [python-visualization/folium#808]
#### Code sample

```python

global m
    
m = folium.Map(location=center, tiles='OpenStreetMap', zoom_start=zoom_lvl, prefer_canvas=True)
    
# folium.plugins.FastMarkerCluster(coords).add_to(m)
# m.add_child(folium.plugins.FastMarkerCluster(coords))
# folium.plugins.MarkerCluster(coords).add_to(m)

for pt in coords :
        
     folium.Marker(pt).add_to(m)

```
#### Problem description

I'm making an interactive visualization map inside a Jupyter Notebook, by displaying some points (avg : 1000-2000, max : 25000). I've tried several ways : 
* placing markers one by one
* clustermarker
* fastclustermarker
Each one present advantages and drawbacks. Making markers one by one is way too slow, clustermarker is less slow and allow me to display popup Vincent graphs as I'd like, but it is too long to generate all the charts and put them in the popup, and fastclustermarker doesn't allow me to display popup charts.

I also thought about getting map zoomed coordinates (lat, long) so as to display points in this area only (great zoom when in need of Vincent chart information), that would make 20 points approximately.
In other words, the Vincent chart only needs to be computed and added to the marker when the zoom over an area is great, because the number of points when zoom is normal is too high for selecting one for information.

Is there any way to : 

* Get markers added to a map for modification
* Get map current bounds (topleft and bottom right e.g.), after movement and zoom, to dynamically display markers only in the zoomed area.
* To add info popup to FastMarkerCluster element

**Comment by Conengmo** [comment]
Generating loose points should be faster now #811 is merged. 

Getting callbacks from your browser back to Python is currently not possible with folium, so you can't dynamically insert Vincent charts based on the zoom area. Of course you can build this yourself by working directly with the Leaflet API and doing javascript callbacks to some server than can serve data. But building full-fledged webapps is outside of the scope for folium.

You could hack FastMarkerCluster to enable different popups for each marker. But I'm not sure if it's worth the trouble.

**Comment by Conengmo** [comment]
I'm closing this old issue since it doesn't seem actionable. If you have new information feel free to add it, we can reopen if needed.
